tree/ntuple/v7/src/RFieldUtils.cxx

```diff
@@ -122,9 +122,9 @@ std::vector<AnglePos> FindTemplateAngleBrackets(const std::string &typeName)
 
 } // namespace
 
-std::string ROOT::Internal::GetCanonicalTypePrefix(const std::string &typeName)
+std::string ROOT::Internal::GetCanonicalTypePrefix(std::string_view typeName)
 {
-   std::string canonicalType{TClassEdit::CleanType(typeName.c_str(), /*mode=*/1)};
+   std::string canonicalType{TClassEdit::CleanType(typeName.data(), /*mode=*/1)};
```

Comment:
`string_view::data()` is not necessarily NULL-terminated. If the function accepts a `std::string_view`, it must create a `std::string` copy in order to then get a correct `c_str()`.